### PR TITLE
correctly determine protocol from port and configuration in PHP

### PR DESCRIPTION
**Pull request by paddatrapper**
The PHP logic now matches the Python. Fixes: #1283

@Robbt would appreciate your review here, PHP is not a strong point for me

### CODE REVIEW

File: airtime_mvc/application/configs/conf.php [hunk 1]
```diff
@@ -32,6 +32,7 @@ public static function loadConfig() {
         $CC_CONFIG['stationId'] = $values['general']['station_id'];
         $CC_CONFIG['phpDir'] = $values['general']['airtime_dir'];
         $CC_CONFIG['forceSSL'] = isset($values['general']['force_ssl']) ? $values['general']['force_ssl'] : FALSE;
```
Comment (jooola) [inline airtime_mvc/application/configs/conf.php:34]: I would reconsider this line, and how it is then used as value in some conditions.
What I understand is that, if anything is set, say `no` (or `yes`, doesn't matter), the `$forceSSL` var will always evaluate as true, and only to false if it isn't set.

This is not the behavior I saw in the python code.

**Review by jooola — COMMENTED** [review]
Again, this is just patching a small problem, I am really not in favor for keeping all this logic in the runtime code.

URL/URI are super powerful to describe all the different variables/specificity we are trying to keep in the config. So I would rather see a `base_url` config entry, with a full URL and eventually explode it if we need some parts of it, than trying to build a proper URL in different part of the code, over and over.

**Comment by paddatrapper** [comment]
> Again, this is just patching a small problem, I am really not in favor for keeping all this logic in the runtime code.

Sure, but I want to get 3.0 out and look at breaking config changes later (maybe with the deprecation of the PHP code).

> URL/URI are super powerful to describe all the different variables/specificity we are trying to keep in the config. So I would rather see a base_url config entry, with a full URL and eventually explode it if we need some parts of it, than trying to build a proper URL in different part of the code, over and over.

There are two places this happens - this function in PHP and in api_clients/utils.py. Everything else should call one of the two. But yeah, a full `base_url` is probably better as I don't see a case where we need an individual portion anywhere

File: airtime_mvc/application/configs/conf.php (hunk 1 above)
Comment (paddatrapper) [reply airtime_mvc/application/configs/conf.php:34]: I don't know PHP well enough to fix this...

Comment (jooola) [reply airtime_mvc/application/configs/conf.php:34]: I'll try to push something

Comment (jooola) [reply airtime_mvc/application/configs/conf.php:34]: See https://github.com/paddatrapper/libretime/pull/14

Comment (paddatrapper) [reply airtime_mvc/application/configs/conf.php]: Merged into this PR

**Comment by jooola** [comment]
@paddatrapper Hey, did you find a better solution or was this change unnecessary ?

**Comment by paddatrapper** [comment]
That was a mistake, re-opened as #1288

**Pull request closed without merging**